config/env.config.js

```diff
@@ -55,5 +55,25 @@ const config = {
     }
 };
 
+const redirect = function(options) { 
+    this.apply = function(compiler) {
+        compiler.plugin('compilation', function(compilation) {
+            compilation.plugin('html-webpack-plugin-before-html-processing', function(htmlPluginData, callback) {
+                let htmlHead = htmlPluginData.html.match('<head>').index;
+                htmlPluginData.html = htmlPluginData.html.slice(0, htmlHead) + '<head>' +
+                    `<script>
+                        var newUrl = window.localStorage.getItem('environmentConfig');
+                        if (newUrl && !window.location.href.match(newUrl)) {
+                            window.location.href = newUrl + window.location.pathname;
+                        }
+                    </script>`
+                    + htmlPluginData.html.slice(htmlHead + 6);
```

Comment:
I'm assuming 6 is the length of `<head>`?  Could you either put it as an inlined comment next to `6`, or just use `'<head>'.length` directly?

    htmlPluginData.html.slice(htmlHead + 6 /* '<head>' */);

or, better IMO

    htmlPluginData.html.slice(htmlHead + '<head>'.length);